src/encoding_implementations/amplitude_encoding.jl

```diff
@@ -0,0 +1,68 @@
+export AmplitudeEncoding
+
+"""
+    AmplitudeEncoding <: Encoding
+    AmplitudeEncoding(minval::Real, maxval::Real; n = 2)
+
+Encoding which discretizes a state vector (an `AbstractVector` of some kind) into its
+"absolute amplitude symbol"  relative to some pre-dedined minimum value `minval` and
```

Comment:
Absolute amplitude symbol is also weird. This encoding _averages_ the elements of the vector; it does not pick out the one that has the largest amplitude. We should state something like this to guide the reader.

"A vector is encoded based on the relative position the mean of the vector has with respect to a predefined minimum and maximum value".